taskcluster/client.py

```diff
@@ -400,11 +400,23 @@ def _makeHttpRequest(self, method, route, payload):
                 )
 
             # Handle non 2xx status code and retry if possible
+            status = response.status_code
             try:
                 response.raise_for_status()
-                if response.status_code == 204:
+                if status == 204:
                     return None
 
+                # raise_for_status only catches status >= 400, but it also
+                # customizes the error message.  Let's use that customized
+                # error message for status >= 400, and craft one for other
+                # non-2xx statuses.
+                if status < 200 or status >= 300:
+                    raise requests.exceptions.RequestException(
```

Comment:
I'm explicitly raising the exception that we're catching in the `except` line below, so we don't have to deal with exceptions non-2xx statuses in more than one place.  Do you still object?